cmd/customd/client/client.go

```diff
@@ -0,0 +1,448 @@
+package client
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/app"
+	"github.com/iov-one/weave/x/sigs"
+	"github.com/pkg/errors"
+	cmn "github.com/tendermint/tendermint/libs/common"
+	tmpubsub "github.com/tendermint/tendermint/libs/pubsub"
+	"github.com/tendermint/tendermint/rpc/client"
+	ctypes "github.com/tendermint/tendermint/rpc/core/types"
+	tmtypes "github.com/tendermint/tendermint/types"
+)
+
+type Header = tmtypes.Header
+type Status = ctypes.ResultStatus
+type GenesisDoc = tmtypes.GenesisDoc
+
+const BroadcastTxSyncDefaultTimeOut = 15 * time.Second
+
+var QueryNewBlockHeader = tmtypes.EventQueryNewBlockHeader
+
+// Client is an interface to interact with bcp
+type Client interface {
```

Comment:
All methods should be documented here. At least a sentence explaining what the method does. Bonus points for information when and why to use it an any interesting detail that the implementation should be aware of.